Assets/Mirror/Examples/TopDownShooter/Scripts/PlayerTopDown.cs

```diff
@@ -0,0 +1,345 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+
+namespace Mirror.Examples.TopDownShooter
+{
+    public class PlayerTopDown : NetworkBehaviour
+    {
+        public readonly static List<PlayerTopDown> playerList = new List<PlayerTopDown>();
+
+        private Camera mainCamera;
+        private CameraTopDown cameraTopDown;
+        private CanvasTopDown canvasTopDown;
+
+        public float moveSpeed = 5f;
+        public CharacterController characterController;
+
+        public GameObject leftFoot, rightFoot;
+        private Vector3 previousPosition;
+        private Quaternion previousRotation;
+
+        [SyncVar(hook = nameof(OnFlashLightChanged))]
+        public bool flashLightStatus = true;
+        public Light flashLight;
+
+        [SyncVar(hook = nameof(OnKillsChanged))]
+        public int kills = 0;
+
+        [SyncVar(hook = nameof(OnPlayerStatusChanged))]
+        public int playerStatus = 0;
+        public GameObject[] objectsToHideOnDeath;
+
+        public float shootDistance = 100f;
+        public LayerMask hitLayers;
+        public GameObject muzzleFlash;
+        public AudioSource soundGunShot, soundDeath, soundFlashLight, soundLeftFoot, soundRightFoot;
+
+#if !UNITY_SERVER
+        public override void OnStartLocalPlayer()
+        {
+            // Grab and setup camera for local player only
+            mainCamera = Camera.main;
+            cameraTopDown = mainCamera.GetComponent<CameraTopDown>();
+            cameraTopDown.playerTransform = this.transform;
+            cameraTopDown.offset.y = 20.0f; // dramatic zoom out once players setup
+            canvasTopDown.playerTopDown = this;
+
+            // We want 3D audio effects to be around the player, not the camera 50 meters in the air
+            // Otherwise it looks weird in-game, trust me
+            mainCamera.GetComponent<AudioListener>().enabled = false;
+            this.gameObject.AddComponent<AudioListener>();
+        }
+#endif
+
+        void Awake()
+        {
+            // Allow all players to run this, they may need it for reference
+            canvasTopDown = GameObject.FindObjectOfType<CanvasTopDown>();
+        }
+
+        public void Start()
+        {
+            playerList.Add(this);
+            print("Player joined, total players: " + playerList.Count);
+#if !UNITY_SERVER
+            if (isClient)
+            {
+                InvokeRepeating("AnimatePlayer", 0.2f, 0.2f);
+            }
+#endif
+        }
+
+        public void OnDestroy()
+        {
+            playerList.Remove(this);
+            print("Player removed, total players: " + playerList.Count);
+            if (mainCamera) { mainCamera.GetComponent<AudioListener>().enabled = true; }
+        }
```

Comment:
playerList is only used on server...add / remove in OnStart|StopServer
```suggestion
        public void Start()
        {
            print("Player joined, total players: " + playerList.Count);
#if !UNITY_SERVER
            if (isClient)
            {
                InvokeRepeating("AnimatePlayer", 0.2f, 0.2f);
            }
#endif
        }

        public void OnDestroy()
        {
            print("Player removed, total players: " + playerList.Count);
            if (mainCamera) { mainCamera.GetComponent<AudioListener>().enabled = true; }
        }
```

Reply:
Oh I did it for player count, "print("Player joined, total players: " + playerList.Count);"
So local players would know how many players are in game, without it being sync var'd or them using a .Find. Although its not displayed in any UI currently, I still think its best to keep it as is, I will add a note though about using OnStartServer/OnStopServer if joining players do not need the list.